packages/www/pages/dashboard/billing/index.tsx

```diff
@@ -63,12 +80,97 @@ const Billing = () => {
       }
     };
 
-    const doGetUsage = async (fromTime, toTime, userId) => {
-      const [res, usage] = await getUsage(fromTime, toTime, userId);
+    const doGetUsage = async (fromTime, toTime, status) => {
+      fromTime = fromTime * 1000;
+      toTime = toTime * 1000;
+
+      if (status === "canceled") {
+        const now = new Date();
+        fromTime = new Date(now.getFullYear(), now.getMonth(), 1).getTime();
+        toTime = now.getTime();
+      }
+
+      let [res, billingUsage] = await getBillingUsage(fromTime, toTime);
+
       if (res.status == 200) {
-        setUsage(usage);
+        if (billingUsage) {
+          setUsage(billingUsage);
+        } else {
+          billingUsage = {
+            TotalUsageMins: 0,
+            DeliveryUsageMins: 0,
+            StorageUsageMins: 0,
+          };
+          setUsage(billingUsage);
+        }
+        doCaculateOverUsage(billingUsage);
+      }
+    };
+
+    const doCaculateOverUsage = async (usage) => {
+      const overusage = await calculateOverUsage(usage);
+      if (overusage) {
+        const oBill = await calculateOverUsageBill(overusage);
+        setOverUsageBill(oBill);
+        let [res, uInvoice] = await getUpcomingInvoice(user.stripeCustomerId);
+        setUpcomingInvoice(uInvoice?.invoices);
+        setUpcomingInvoiceTotal((uInvoice?.invoices.total / 100) | 0);
       }
     };
+
+    const calculateOverUsage = async (usage) => {
```

Comment:
Is this the same code from the other component? If it's not easy to re-use the backend logic, can you move it to a helper and avoid repeating it between the 2 frontend components?